### PR TITLE
fix infinite loop on redirect to file

### PR DESCRIPTION
**Pull request by mrkane27**
Fix https://github.com/nuget/home/issues/1893

### CODE REVIEW

**Comment by yishaigalatzer** [comment]
Sorry - Commented on the issue instead of the PR.

Thanks for this cool fix. We would love to take it after adding a failing test in the current code, that gets fixed by this change.

**Comment by mrkane27** [comment]
Thanks for replying!

As far as I can tell, the issue only happens on Linux and Mac, using Mono.
Do you run tests on any of those? If yes, could you point me to where I should add a test? -- I've never contributed to this repository before.

**Comment by yishaigalatzer** [comment]
Ah - Cool.

We are not yet running tests on mac and linux regularly (but in fact that is what half the team is doing this week, making it all happen for core clr).

Lets do this for now
1. Add a comment that this only happens on mono
2. We will try that with the new nuget.commandline.xplat package (to make sure it doesn't repeat there)
3. Add a follow up test, once we have the linux tests passing properly (we are still tracking a few tens of test failures)

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/ConsoleTest.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+using Xunit;
+
+namespace NuGet.CommandLine.Test
+{
+    public class ConsoleTest
+    {
+        [Fact]
+        public void TestConsoleWindowWidthNotZero()
+        {
+            Assert.NotEqual(0, new NuGet.Common.Console().WindowWidth);
```
Comment (mrkane27) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/ConsoleTest.cs:12]: This test will only be relevant if someone runs the tests on Linux, with output redirect. I tried messing around with `System.Console.SetOut()` and the like at runtime, but Mono does some [initialization](https://goo.gl/dLCVCa) before that, so it won't apply.

**Comment by mrkane27** [comment]
A different take might be to use `System.Console.IsOutputRedirected` and default to e.g. a width of 80, which would make sense for file output -- but that might be overkill for this fix.

**Comment by yishaigalatzer** [comment]
I like this new idea better. Lets update to it or does not sound like an overkill.

We are running tests on Linux on a CI regularly starting from this week.

**Comment by mrkane27** [comment]
I thought about this for a bit more, and it might not actually work. For example, if you do `nuget ... | cat`, you might still want it formatted for the terminal. I don't think there's a reliable way to determine whether the final output is into a file. What do you think?

I've also run some [tests](https://gist.github.com/mrkane27/b1d8fe9de43810591aac) on how `Console` behaves on .NET and Mono, if you want to take a look.

**Comment by yishaigalatzer** [comment]
I think the key is to unblock the hang. In general we are moving NuGet off mono and moving it gradually to coreCLR on unix (see nuget.commandline.xplat package), so coming up with something reasonable (like your suggestion of 80 chars) might be good enough. And then on coreCLR we can try to get things fixed.

**Comment by mrkane27** [comment]
Is this current state what you had in mind?

**Comment by deepakaravindr** [comment]
@mrkane27, Thanks for the tests on how Console behaves on .NET and Mono. I have checked it out. Your fix looks good to me too. I have merged it with https://github.com/NuGet/NuGet.Client/commit/6afc0acb41be2a316dff3ca38d1b15cb75b0e0d4

**Pull request closed without merging**